### ISSUE TITLE
Return Error Messages - AWS mostly gives 400 error codes with message in the body.

### ISSUE THREAD

**Issue by rupertlssmith** [the-sett/elm-aws-core#2]
The send functions look like:

```elm
send : Service -> Credentials -> Request a -> Task Error a
```
 
Which only returns `Http.Error` in the event of an error, and this only gives `BadStatus 400` in the event of an AWS error. Would be better to define an `Error` type specifically for this package, that also gives back the AWS errors.

Workaround: Use https://package.elm-lang.org/packages/the-sett/elm-aws-core/latest/AWS-Core-Http#setResponseParser

**Comment by rupertlssmith** [comment]
`Http.Error` is:

```elm
type Response body
    = BadUrl_ String
    | Timeout_
    | NetworkError_
    | BadStatus_ Metadata body
    | GoodStatus_ Metadata body
```

And this is reduced to `Http.Error` which is:

```elm
type Error
    = BadUrl String
    | Timeout
    | NetworkError
    | BadStatus Int
    | BadBody String
```

Propsed `AWS.Core.Http.Error` might look like:

```elm
type Error
    = BadUrl String
    | Timeout
    | NetworkError
    | BadStatus Int
    | BadBody String
    | AWSError Int String
```

Where the 400 responses with a message get turned into `AWSError 400 "Something went wrong". `BadStatus` would still be there for capturing the 500 error codes.

Need to look in a bit more detail into exactly what error messages AWS can return? Is it always 400 and a string error message?

What about the REST-JSON APIs, do they use a range of status codes or the same everything is a 400 approach? Are the error messages more complex, and require decoders? 

What about the XML based APIs, do they return error messages in XML format?

Maybe need to define an `AWSErrorMessage` type which captures all the possibilities.

**Comment by rupertlssmith** [comment]
The actual error type I went for is:

```elm
type Error err
    = HttpError Http.Error
    | AWSError err
```

And the caller can specify a decoder for `err`.

This is being released as version 8.0.
